src/Arcus.WebApi.Jobs/CloudEventBackgroundJob.cs

```diff
@@ -0,0 +1,136 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.ServiceBus;
+using CloudNative.CloudEvents;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Management;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Arcus.WebApi.Jobs
+{
+    /// <summary>
+    /// Representing a Azure Service Bus Topic message pump that will create and delete a Service Bus Topic subscription during the lifetime of the pump.
+    /// </summary>
+    public abstract class CloudEventBackgroundJob : AzureServiceBusMessagePump<CloudEvent>
+    {
+        private static readonly JsonEventFormatter JsonEventFormatter = new JsonEventFormatter();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloudEventBackgroundJob"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration of the application.</param>
+        /// <param name="serviceProvider">The collection of services that are configured.</param>
+        /// <param name="options">The options to further configure this job.</param>
+        /// <param name="logger">The logger to write telemetry to.</param>
+        /// <exception cref="ArgumentNullException">When the <paramref name="serviceProvider"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">When the <paramref name="serviceProvider"/> doesn't have a registered <see cref="AzureServiceBusMessagePumpSettings"/> instance.</exception>
+        protected CloudEventBackgroundJob(
+            IConfiguration configuration,
+            IServiceProvider serviceProvider,
+            IOptions<CloudEventBackgroundJobOptions> options,
+            ILogger logger) : base(configuration, serviceProvider, logger)
+        {
+            Guard.NotNull(
+                serviceProvider,
+                nameof(serviceProvider),
+                $"Requires a '{nameof(IServiceProvider)}' implementation to retrieve the '{nameof(AzureServiceBusMessagePumpSettings)}'");
+            Guard.NotNull(options, nameof(options), $"Requires a '{nameof(IOptions<CloudEventBackgroundJobOptions>)}' to correctly configure this job");
+            Guard.For<ArgumentException>(() => options.Value is null, $"Requires a '{nameof(IOptions<CloudEventBackgroundJobOptions>)}' to correctly configure this job");
+
+            var messagePumpSettings = serviceProvider.GetRequiredService<AzureServiceBusMessagePumpSettings>();
+            Guard.NotNull<AzureServiceBusMessagePumpSettings, ArgumentException>(
+                messagePumpSettings, 
+                $"The '{nameof(serviceProvider)}:{serviceProvider.GetType().Name}' requires to have a non-null '{nameof(AzureServiceBusMessagePumpSettings)}' instance registered");
+            
+            JobId = options.Value.JobId;
```

Comment:
Thinking out loud - What's the benefit of moving this to the options? Shouldn't this be something we manage?

Reply:
Yes, that was what I commented in your PR on the messaging repo.
The problem exists because the `AzureServiceBusMessagingOptions.SubscriptionName` is readonly and is set during the registration of hosted service.

So unless you want me to substing it from the `.SubscriptionName`; I have to find a way to set the subscription name with the `JobId` beforehand; meaning I have to get another way to get the `JobId` in the `CloudEventBackgroundJob` ==> dedicated options.

Reply:
Personally I think the option should be the prefix and not the full subscription name, Arcus will determine that.

Once the job starts, it will generate a `JobId` and append it to the prefix to define the subscription id to use.

So `AzureServiceBusMessagingOptions.SubscriptionName` becomes `AzureServiceBusMessagingOptions.SubscriptionNamePrefix` and rest of logic is in constructor.

Reply:
Ok, but then the 'main' message pump should have access to the 'full' subscription name, no? Otherwise it will use the prefix only.

Reply:
Oh now I get why it's like this. There's no way around it indeed as the underlying pump has to know it already.

Sorry, I was confused.